sparta/sparta/resources/Pipeline.hpp

```diff
@@ -266,22 +296,39 @@ namespace sparta
 
             // Create a new stage event handler, and add it to its event list
             auto & event_list = event_list_at_stage_[id];
-            event_list.emplace_back(new sparta::UniqueEvent<sched_phase>(&es_,
-                                        "uev_stage_" + std::to_string(id) + "_" + std::to_string(event_list_at_stage_[id].size()),
-                                        handler));
+            if constexpr (std::is_same_v<EventT, PhasedPayloadEvent<DataT>>) {
+                event_list.emplace_back(new EventT(es_,
```

Comment:
Add an assert to make sure `handler.argCount() == 1`

Reply:
Thanks!